cmd/kubeadm/app/cmd/config.go

```diff
@@ -118,36 +116,80 @@ func NewCmdConfigPrintDefault(out io.Writer) *cobra.Command {
 	return cmd
 }
 
 func getDefaultAPIObjectBytes(apiObject string) ([]byte, error) {
-	var internalcfg runtime.Object
-	var err error
 	switch apiObject {
 	case constants.InitConfigurationKind, constants.MasterConfigurationKind:
-		internalcfg, err = configutil.ConfigFileAndDefaultsToInternalConfig("", &kubeadmapiv1alpha3.InitConfiguration{
-			API:               kubeadmapiv1alpha3.API{AdvertiseAddress: "1.2.3.4"},
-			BootstrapTokens:   []kubeadmapiv1alpha3.BootstrapToken{sillyToken},
-			KubernetesVersion: fmt.Sprintf("v1.%d.0", constants.MinimumControlPlaneVersion.Minor()+1),
-		})
+		return getDefaultInitConfigBytes()
+
 	case constants.NodeConfigurationKind:
-		internalcfg, err = configutil.NodeConfigFileAndDefaultsToInternalConfig("", &kubeadmapiv1alpha3.NodeConfiguration{
-			Token: sillyToken.Token.String(),
-			DiscoveryTokenAPIServers:               []string{"kube-apiserver:6443"},
-			DiscoveryTokenUnsafeSkipCAVerification: true,
-		})
-		// TODO: DiscoveryTokenUnsafeSkipCAVerification: true needs to be set for validation to pass, but shouldn't be recommended as the default
+		return getDefaultNodeConfigBytes()
+
 	default:
-		err = fmt.Errorf("--api-object needs to be one of %v", availableAPIObjects)
+		// Is this a component config?
+		registration, ok := componentconfigs.Known[componentconfigs.RegistrationKind(apiObject)]
+		if !ok {
+			return []byte{}, fmt.Errorf("--api-object needs to be one of %v", getAvailableAPIObjects())
+		}
+		return getDefaultComponentConfigBytes(registration)
+	}
+}
+
+func getAvailableAPIObjects() []string {
+	objects := []string{constants.InitConfigurationKind, constants.MasterConfigurationKind, constants.NodeConfigurationKind}
+	for componentType := range componentconfigs.Known {
+		objects = append(objects, string(componentType))
 	}
+	return objects
+}
+
+func getDefaultedInitConfig() (*kubeadmapi.InitConfiguration, error) {
+	return configutil.ConfigFileAndDefaultsToInternalConfig("", &kubeadmapiv1alpha3.InitConfiguration{
+		API:               kubeadmapiv1alpha3.API{AdvertiseAddress: "1.2.3.4"},
+		BootstrapTokens:   []kubeadmapiv1alpha3.BootstrapToken{sillyToken},
+		KubernetesVersion: fmt.Sprintf("v1.%d.0", constants.MinimumControlPlaneVersion.Minor()+1),
+	})
+}
+
+func getDefaultInitConfigBytes() ([]byte, error) {
+	internalcfg, err := getDefaultedInitConfig()
 	if err != nil {
 		return []byte{}, err
 	}
+
 	return configutil.MarshalKubeadmConfigObject(internalcfg)
 }
 
+func getDefaultNodeConfigBytes() ([]byte, error) {
+	internalcfg, err := configutil.NodeConfigFileAndDefaultsToInternalConfig("", &kubeadmapiv1alpha3.NodeConfiguration{
+		Token: sillyToken.Token.String(),
+		DiscoveryTokenAPIServers:               []string{"kube-apiserver:6443"},
+		DiscoveryTokenUnsafeSkipCAVerification: true, // TODO: DiscoveryTokenUnsafeSkipCAVerification: true needs to be set for validation to pass, but shouldn't be recommended as the default
```

Comment:
oh, printing this as a default to `true` might not be a good idea...
i assume it's possible to modify it in the `internalcfg` before marshaling?
+ fix the validation eventually too, i guess.

Reply:
I understand it, but this is out of the scope for this PR. I can try to get a workaround for that, but this can be done in another PR.